spring-cloud-dataflow-server-local/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java

```diff
@@ -0,0 +1,134 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+import javax.servlet.RequestDispatcher;
+
+import static org.springframework.restdocs.headers.HeaderDocumentation.headerWithName;
+import static org.springframework.restdocs.headers.HeaderDocumentation.responseHeaders;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.springframework.restdocs.JUnitRestDocumentation;
+
+/**
+ * @author Gunnar Hillert
+ */
+public class ApiDocumentation extends BaseDocumentation {
+
+	@ClassRule
+	public static TestRule springDataflowServer = localDataflowResource;
+
+	@Rule
+	public JUnitRestDocumentation restDocumentation =
+			new JUnitRestDocumentation(TARGET_DIRECTORY);
+
+	@Before
+	public void setupMocks() {
+		super.prepareDocumentationTests(restDocumentation);
+	}
+
+	@Test
+	public void headers() throws Exception {
+		this.mockMvc
+			.perform(get("/"))
+			.andExpect(status().isOk())
+			.andDo(print())
+			.andDo(this.documentationHandler.document(
+				responseHeaders(
+					headerWithName("Content-Type").description("The Content-Type of the payload, e.g. `application/hal+json`"))));
+	}
+
+	@Test
+	public void errors() throws Exception {
+		this.mockMvc
+			.perform(get("/error")
+				.requestAttr(RequestDispatcher.ERROR_STATUS_CODE, 400)
+				.requestAttr(RequestDispatcher.ERROR_REQUEST_URI, "/apps")
+				.requestAttr(RequestDispatcher.ERROR_MESSAGE, "The app 'http://localhost:8080/apps/123' does not exist"))
+			.andDo(print())
+			.andExpect(status().isBadRequest())
+			.andExpect(jsonPath("error", is("Bad Request")))
+			.andExpect(jsonPath("timestamp", is(notNullValue())))
+			.andExpect(jsonPath("status", is(400)))
+			.andExpect(jsonPath("path", is(notNullValue())))
+			.andDo(this.documentationHandler.document(
+				responseFields(
+					fieldWithPath("error").description("The HTTP error that occurred, e.g. `Bad Request`"),
+					fieldWithPath("message").description("A description of the cause of the error"),
+					fieldWithPath("path").description("The path to which the request was made"),
+					fieldWithPath("status").description("The HTTP status code, e.g. `400`"),
+					fieldWithPath("timestamp").description("The time, in milliseconds, at which the error occurred"))));
+	}
+
+	@Test
+	public void index() throws Exception {
+		this.mockMvc.perform(get("/"))
+			.andExpect(status().isOk())
+			.andDo(print())
+			.andDo(this.documentationHandler.document(
+				links(
```

Comment:
The descriptions in here use different forms: sometimes verbs, sometimes nouns. Sometimes they state the obvious, sometimes they add more context. We should be consistent IMHO